agent/internal/actor/action.go

```diff
@@ -40,6 +44,24 @@ func (a blockAction) ActionID() string {
 	return a.ID
 }
 
+type redirectAction struct {
+	ID, URL string
+}
+
+func newRedirectAction(id, location string) (*redirectAction, error) {
+	if _, err := url.ParseRequestURI(location); err != nil {
+		return nil, sqerrors.Wrap(err, "validation of the redirection location url")
```

Comment:
Include a "rawly serialised" (something like `#inspect` in Ruby, IIRC that's `%v` or something in Go) erroring value in logs. This will help *a lot* when things go south.

Reply:
It present in the returned error. For example `parse http//sqreen.com/: invalid URI for request` (https://play.golang.org/p/mGoby5m5Jgj)